core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java

```diff
@@ -192,12 +192,23 @@ private void compressDataMyOwnWay(ColumnWithShortIndex[] indexes) {
     }
     map.add(start);
     map.add(counter);
-    this.keyBlock = convertToKeyArray(list);
-    if (indexes.length == keyBlock.length) {
-      dataIndexMap = new short[0];
-    } else {
+    boolean useRle = (((list.size() + map.size()) * 100) / indexes.length) > 70 ? false : true;
```

Comment:
Can we decide this in a more heuristic way? Like if we find there are more than 5 pages not doing RLE, then do not pay the cost to try to compress it in all future blocklets.